core/src/main/java/io/grpc/internal/TransportSet.java

```diff
@@ -106,22 +111,23 @@
   private boolean shutdown;
 
   /**
-   * The future for the transport for new outgoing requests. 'lock' lock must be held when assigning
-   * to activeTransportFuture.
+   * The future for the transport for new outgoing requests. 'lock' must be held when assigning
+   * to it.
    */
+  @Nullable
   private volatile SettableFuture<ClientTransport> activeTransportFuture;
 
   TransportSet(EquivalentAddressGroup addressGroup, String authority, LoadBalancer loadBalancer,
       BackoffPolicy.Provider backoffPolicyProvider, ClientTransportFactory transportFactory,
-      ScheduledExecutorService scheduledExecutor, Callback callback) {
+      ScheduledExecutorService scheduledExecutor, Ticker ticker, Callback callback) {
```

Comment:
Hmm... `Ticker` is `@Beta`. Just pass in a `Stopwatch` directly?

Note that we can avoid injecting the `Stopwatch` from `ManagedChannelImpl` by having something like:
```java
  TransportSet(EquivalentAddressGroup addressGroup, String authority, LoadBalancer loadBalancer,
      BackoffPolicy.Provider backoffPolicyProvider, ClientTransportFactory transportFactory,
      ScheduledExecutorService scheduledExecutor, Callback callback) {
    this(addressGroup, authority, loadBalancer, backoffPolicyProvider, transportfactory, scheduledExecutor,
        Stopwatch.createUnstarted(), callback);
  }
```

And then we are able to make this constructor `@VisibleForTesting` so that users of the class don't need to care about injection intended for Testing. In this case it may not be worth it though.